src/js/common/lib/i18n.js

```diff
@@ -0,0 +1,188 @@
+/*
+
+    Functions to assist in internationalising and localising content.
+
+*/
+/* eslint-env node */
+"use strict";
+var fluid = fluid || require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+fluid.registerNamespace("gpii.handlebars.i18n");
+
+/**
+ *
+ * Parse the content of an `Accept-Language` header and return the full list of locales in order of preference.
+ *
+ * @param {String} header - The value of the `Accept-Language` header.
+ * @return {String[]} - An array of strings representing locales and/or languages, in order of preference.
+ *
+ */
+gpii.handlebars.i18n.getAllLocalesFromHeader = function (header) {
+    var acceptedLanguages = [];
+    if (header === undefined) {
+        acceptedLanguages.push("*");
+    }
+    else if (typeof header === "string") {
+        var localeSegments = header.trim().split(/ *, */);
+        var segmentsAsObjects = [];
+        fluid.each(localeSegments, function (localeSegment) {
+            var subSegments = localeSegment.split(";q=");
+            var segmentDef = {
+                locale: subSegments[0],
+                q: subSegments.length > 1 ? subSegments[1] : 1
+            };
+            segmentsAsObjects.push(segmentDef);
+        });
+        segmentsAsObjects.sort(gpii.handlebars.i18n.sortByQScore);
+        segmentsAsObjects.map(function (entry) {
+            acceptedLanguages.push(entry.locale);
+        });
+    }
+    return acceptedLanguages.map(function (entry) { return entry.toLowerCase().replace("-", "_");});
+};
+
+/**
+ *
+ * Sort by the "q score" included with complex "Accept-Language" headers.
+ *
+ * @param {Object} a - A single weighted language entry.
+ * @param {Object} b - Another weighted language to compare to `a`.
+ * @return {Number} - -1 if `a` should come before `b`, 1 if `b` should come before `a`, 0 if their weights are equal.
+ *
+ */
+gpii.handlebars.i18n.sortByQScore = function (a, b) {
+    if (a.q === b.q) {
```

Comment:
This can simply read 
return b.q - a.q;

Reply:
Note that because you used triple equals here, the comparator would have stuffed up in the case that one q value was "1" and the other was 1

Reply:
Well, the stuff would have been unobservable since the ordering should be indeterminate, but all the same ..... : P

Reply:
Simplified and also included a parseInt call on both sides of the comparison.

Reply:
I actually followed up on your later suggestion and parsed it elsewhere to keep the function simple.